### PR TITLE
Support partition based incremental policy

### PR DESCRIPTION
**Pull request by wangzhen11aaa**
https://github.com/AlexAkulov/clickhouse-backup/issues/349

### CODE REVIEW

**Comment by wangzhen11aaa** [comment]
/review.

File: pkg/backup/backup.go [hunk 1]
```diff
@@ -283,7 +283,7 @@ func AddTableToBackup(ch *clickhouse.ClickHouse, backupName string, diskList []c
 	}
 
 	// backup data
-	if !strings.HasSuffix(table.Engine, "MergeTree") {
+	if !(strings.HasSuffix(table.Engine, "MergeTree")) {
```
Comment (Slach) [inline pkg/backup/backup.go]: please, remove unnecessary brackets here

File: pkg/backup/upload.go [hunk 2]
```diff
@@ -305,22 +306,39 @@ func (b *Backuper) markDuplicatedParts(backup *metadata.BackupMetadata, existsTa
...
+					if err := isDuplicatedParts(existsPath, newPath); err != nil {
+						apexLog.Debugf("part '%s' and '%s' must be the same: %v", existsPath, newPath, err)
+						continue
+					}
+					// Required : this part is duplicated with some part of existTable.
+					newParts[i].Required = true
+				}
+			} else {
+				for i := range newParts {
+					if _, ok := existsPartitionMap[strings.Split(newParts[i].Name, "_")[0]]; ok {
```
Comment (Slach) [inline pkg/backup/upload.go:338]: Fundamentally wrong algorithm
You can't ensure the "new" partition for only new partition name
you should try to learn how exactly work background merges in MergeTree

**Review by Slach — CHANGES_REQUESTED** [review]
please, look review comments

and my comment on issue https://github.com/AlexAkulov/clickhouse-backup/issues/349#issuecomment-1008833520

File: pkg/backup/backup.go (hunk 1 above)
Comment (wangzhen11aaa) [reply pkg/backup/backup.go]: Ok

File: pkg/backup/upload.go (hunk 2 above)
Comment (wangzhen11aaa) [reply pkg/backup/upload.go:338]: Hi, That is just another policy for partition based incremental.

Comment (Slach) [reply pkg/backup/upload.go:338]: try to think again

**Pull request closed without merging**